content/applications/finance/accounting/fiscal_localizations/localizations/india.rst

```diff
@@ -187,3 +190,152 @@ website.
+
+   .. image:: india/e-waybill-gsp-registration.png
+      :align: center
+      :alt: E-Waybill GSP options
+
+#. Enter the OTP code and click on :guilabel:`Verify OTP`.
+
+   .. image:: india/verify-otp.png
+      :align: center
+      :alt: E-Waybill verify otp
+
+#. Check If :guilabel:`Tera Software Limited` is already in List of GSP/ERP Registered.If it is already in list
+   then use that username and password else continue to next steps.
+
+   .. image:: india/e-waybill-gsp-list.png
+      :align: center
+      :alt: E-Waybill added GSP
+
+#. Select :guilabel:`Add/New`, select :guilabel:`Tera Software Limited` as your GSP Name, and create
+   :guilabel:`Username` and :guilabel:`Password` for your API.
+
+   .. image:: india/submit-gsp-api-registration-details.png
+      :align: center
+      :alt: Submit GSP api registration details
+
+#. Click on :guilabel:`Add`.
+
+
+.. _india/e-waybill-configuration:
+
+Configuration on Odoo
+---------------------
+
+To set up the E-Waybill service, go to :menuselection:`Accounting --> Configuration --> Settings -->
+Indian Electronic WayBill`, and enter the :guilabel:`Username` and :guilabel:`Password`.
+
+.. image:: india/e-waybill-setup.png
+   :align: center
+   :alt: E-Waybill setup odoo
+
+.. _india/e-waybill-workflow:
+
+Workflow
+--------
+
+To start E-waybill from Odoo, an invoice/bill must be created with details of E-waybill using the standard
+invoicing/bill flow, that is either from a sales/purchase order or the invoice/bill menu in the Accounting
+application.
+
+.. _india/e-waybill-send:
+
+Send E-waybill
+~~~~~~~~~~~~~~
+
+.. important::
+   If E-waybill (IN) is selected in, :ref:`Sale/Purchase Journal <india/e-invoicing-journals>` then E-waybill will
+   be sent automatically when the invoice/bill is confirmed.
+   If you don't want to send E-waybill when confirming the invoice/bill, don't select E-waybill (IN).
+
+There will be button on invoice/bill to send E-Waybill.
+
+.. image:: india/invoice-send-e-waybill-button.png
+   :align: center
+   :alt: Invoice send E-waybill button
+
+.. _india/invoice-validation-e-way:
+
+Invoice validation
+~~~~~~~~~~~~~~~~~~
+
+Once the invoice is validated and :guilabel:`SEND E-WAYBILL` is clicked, a confirmation message will be
+displayed at the top.
+
+Odoo automatically uploads the JSON-signed file to the government portal after a while. If you want
+to process the invoice immediately, you can click on :guilabel:`Process Now`.
+
+.. image:: india/e-waybill-process.png
+   :align: center
+   :alt: Indian e-waybill confirmation message: "The invoice will be processed asynchronously by
+         the following E-waybill service : E-waybill (IN)"
+
+.. note::
+   - You can find the JSON-signed file in the attached files, in the chatter.
+
+
+Invoice PDF Report
+~~~~~~~~~~~~~~~~~~
+
+Once the E-waybill is submitted, you can print the invoice PDF report. The report includes the
+E-waybill Number and E-waybill validity date.
```

Comment:
It is weird: so I send e-invoice then eway bill, but then send the e-invoice with the ewaybill number?  (should not we make distinction between the 2 flows: when you need IRN and not?)  In general, it would be good to also more describe the flow: when you send goods, you typically validate the picking first, then from the sale order you create the invoice based on delivered qties, ...  (just to describe a typical flow next to how it can be done directly from the invoice)

Reply:
if the customer is eligible for E-invoice and the invoice is B2B or export then without IRN E-waybill is not created.
so here we consider normal E-waybill flow without IRN.

we don't want to confuse users by taking picking in the flow because technically no direct relation between picking and invoice.

instead of just telling add a message like when you transport goods at that time create E-waybill.
and By default, all businesses know what is waybill and when they need to create it.
because all businesses already follow that.